swap/swap.go

```diff
@@ -314,22 +314,52 @@ func createOwner(prvkey *ecdsa.PrivateKey) *Owner {
 	}
 }
 
-// Add is the (sole) accounting function
-// Swap implements the protocols.Balance interface
-func (s *Swap) Add(amount int64, peer *protocols.Peer) (err error) {
+// checkBalance checks that the amount would not result in crossing the disconnection threshold
+func (s *Swap) checkBalance(amount int64, peer *protocols.Peer) (err error) {
```

Comment:
Nitpicking: you are not just checking the balance here, but checking whether you are allowed to add to the balance. I suggest a name that reflects this such as: `checkBalanceAddition`

Reply:
The purpose of calling it like this is that the focus is on checking that the balance is ok; it is irrelevant what and how the balance is checked (in fact I never liked the `Add` name, which I did not choose). 

If something, I would rename it to `checkBalanceOk` or `isBalanceChangeOk` or `isBalanceChangeAllowed` or something like that, but not adding implementation details to the name.

Reply:
`checkBalanceOk` sounds like an improvement to me!